.github/workflows/test.yml

```diff
@@ -26,6 +26,8 @@ jobs:
 
   testint:
     runs-on: ubuntu-latest
+    # only run if not a fork PR targeting main repo due to permissions
```

Comment:
Fwiw I am not sure this comment is any more clear than the code lol. Just to make sure I understand what you are saying here:

This checks to see if you are making a fork targeting *your fork*, not this repo?

If so, that seems like a reasonable middle ground to allow testing without leaking secrets.